dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,144 @@
+from typing import List
+import numpy
+from dfastmi.kernel.typehints import Vector
+
+
+class BedLevelCalculator:
+    """
+    This class is used to calculate bed level related values.
+    """
+
+    def __init__(self, number_of_periods: int):
+        """
+        number_of_periods : int
+         Amount of the equilibrium bed level change for each respective discharge period available.
+        """
+        self.number_of_periods = number_of_periods
+
+    def get_element_wise_maximum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise maximum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmax : numpy.ndarray
+            Maximum bed level change.
+        """
+        return numpy.maximum.reduce(dzb)
+
+    def get_element_wise_minimum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise minimum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmin : numpy.ndarray
+            minimum bed level change.
+        """
+        return numpy.minimum.reduce(dzb)
+
+    def get_linear_average(self, number_of_days: Vector, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the linear average from the given bed levels.
+
+        Arguments
+        ---------
+        number_of_days : Vector
+            A vector of periods indicating the number of days during which each discharge applies.
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzgem : numpy.ndarray
+            Yearly mean bed level change.
+        """
+        dzgem = 0
+        for i in range(self.number_of_periods):
+            dzgem += dzb[i] * (number_of_days[i] + number_of_days[i-1]) / 2
+        return dzgem
+
+    def get_bed_level_changes(self, dzq: List[numpy.ndarray], rsigma: Vector) -> List[numpy.ndarray]:
+        """
+        This routine computes the bed level changes.
+        This routine requires that dzq and rsigma have the same length.
+        A stagnant period can be represented by a period with rsigma = 1.
+
+        Arguments
+        ---------
+        dzq : List[numpy.ndarray]
+            A list of arrays containing the equilibrium bed level change for each respective discharge period.
+        rsigma : Vector
+            A tuple of relaxation factors, one for each period.
```

Comment:
The typehint says it's a `Vector` but in the description it is just a tuple?

Reply:
The Vector and BoolVector are more arrays, but if you check the typehints.py you can see that they are currently stated as tuple[float,...]

Reply:
discussed, no changes, ready to resolve.